mod_app/models/film_model.py

```diff
@@ -22,164 +34,129 @@ def __str__(self):
 
     synopsis = models.TextField(blank=True)
 
-    source_material = models.OneToOneField(
-        Link,
+    source = models.ManyToManyField(
+        Source,
         help_text="Link to the source material",
         blank=True,
-        null=True,
-        on_delete=models.SET_NULL,
-        related_name="source_material_link",
+        limit_choices_to={"is_source": True},
+        related_name="source_link",
     )
     genre = models.ManyToManyField(Tag, related_name="genres", blank=True)
+
     bfi_category = models.CharField(
         max_length=100, blank=True, null=True
     )  # can use choices if preset
 
-    actors = models.ManyToManyField(
-        "Actor", related_name="films", blank=True, verbose_name="Cast"
+    cast = models.TextField(
+        blank=True,
+        null=True,
     )
-    crew = models.ManyToManyField("Crew", related_name="films", blank=True)
+    crew = models.TextField(blank=True, null=True, verbose_name="Credits")
 
-    intertitle_text = models.CharField(max_length=255, blank=True, null=True)
-    intertitle_photo = models.ImageField(
-        upload_to="images/intertitles/", blank=True, null=True
+    video = models.ForeignKey(
+        FileLink,
+        help_text="Link or upload the video file",
+        blank=True,
+        null=True,
+        on_delete=models.SET_NULL,
+        related_name="video_link",
     )
 
     # Technical section
 
     duration = models.IntegerField(
         blank=True,
         null=True,
-        help_text="Enter the run time in minutes.",
+        help_text="Enter the original run time in minutes.",
     )
     current_length = models.IntegerField(
         blank=True,
         null=True,
-        help_text="Enter the run time in minutes.",
+        help_text="Enter the run time of the BFI copy in minutes.",
     )
-
-    ELEMENT_CHOICES = [
-        ("pos", "Scene positive"),
-        ("ctn", "Negative coutertype"),
-        ("intn", "Internegative"),
-        ("lav", "Intermediate positive scene (lavender)"),
-        ("olay", "Titles"),
-    ]
-
-    element = models.CharField(
-        max_length=4,
-        choices=ELEMENT_CHOICES,
-        default="pos",
-    )
-
     support = models.CharField(
         max_length=1,
-        choices=[("S", "Safety"), ("N", "Nitrate")],
-        default="S",
-    )
-
+        choices=[("V", "Viewable"), ("M", "Master")],
+        default="V",
+    )
+    FORMAT_CHOICES = {
+        (9.5, "9.5 mm"),
+        (16, "16 mm"),
+        (35, "35 mm"),
+        (70, "70 mm"),
+    }
```

Comment:
Is this just here in case you need it later? Best to comment it out for now if so.

Reply:
ah - next PR utilises it properly, I'd just forgotten to set the choices in this one